infra/src/main/resources/db/migration/prod/V91__remove_planned_and_future_existing_states.sql

```diff
@@ -0,0 +1,96 @@
+alter table layout.location_track disable trigger version_row_trigger;
+alter table layout.location_track disable trigger version_update_trigger;
+alter table layout.km_post disable trigger version_row_trigger;
+alter table layout.km_post disable trigger version_update_trigger;
+alter table layout.track_number disable trigger version_row_trigger;
+alter table layout.track_number disable trigger version_update_trigger;
+alter table layout.switch disable trigger version_row_trigger;
+alter table layout.switch disable trigger version_update_trigger;
+
+-- Drop constraints, views and functions that reference layout.state and layout.state_category. They will be recreated
+-- by repeatable migrations
+alter table layout.switch drop constraint if exists switch_unique_official_name;
+alter table layout.location_track drop constraint if exists location_track_unique_official_name;
+
+drop view if exists layout.switch_change_view;
+drop view if exists layout.location_track_change_view;
+drop view if exists layout.km_post_change_view;
+drop view if exists layout.track_number_change_view;
+
+drop view if exists layout.switch_publication_view;
+drop view if exists layout.location_track_publication_view;
+drop view if exists layout.km_post_publication_view;
+drop view if exists layout.track_number_publication_view;
```

Comment:
Ikävä sivuvaikutus sille että repeatable-migraatiot ajetaan viimeisenä: Näitä näkymiähän ei enää oikeasti varmaankaan ole olemassa silloin kun tämä migraatio viedään produun, mutta ainakin siinä dumpissa mitä itse käytän (päivätty 23.5.2024) ne vielä löytyvät. Näille näkymille on poistomigraatio repeatablejen puolella, mutta koska repeatablet ajetaan vasta versioitujen migraatioiden jälkeen, se ei ehdi ajautua ennen tätä ja näkymät ovat olemassa kun tämä migraatio ajetaan.

Näistä viitattiin `state`- ja `state_category`-columneihin, joten jos ne ovat olemassa ja niitä ei poista niin tämä migraatio ei mene läpi. Kannalla josta ne on jo poistettu nämä lauseethan eivät tee mitään koska `if exists`.

Reply:
Joo, toki dev/test/prod kannoissa ne poistot on jo ajettu, mutta varmempi vielä poistaa täällä. Noin periaatteellisella tasolla meidän ei minusta tarvitse varautua siihen että jossain ajettaisi versiota 1.0 vaan riittää että homma toimii viimeisimmän tuotantoversion päälle. Mutta tämä silti ihan hyvä politiikka koska ei tässä maksa mitään.